src/axolotl/datasets_work_queue.py

```
@@ -0,0 +1,382 @@
+                pbar.update(1)
+
+                # Update rate every second
+                current_time = time.time()
+                if current_time - last_update > 1:
+                    elapsed = current_time - start_time
+                    rate = completed / elapsed if elapsed > 0 else 0
+                    pbar.set_postfix({"examples/s": f"{rate:.1f}"})
+                    last_update = current_time
+
+            except queue.Empty:
+                LOG.error("Timeout waiting for results")
```

Ruff (0.15.2)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)